### DISCUSSION TITLE
Can useContractRead return a struct. array, or non-typical return value?

### DISCUSSION THREAD

**Discussion by stackaccount1** [wevm/wagmi#932] [Q&A]
The documentation does not specify or note what this function is capable of returning. Thank you for your response and consideration.

**Comment by dev-cz** [comment]
Yes it can I return various structs and arrays. If you add piece like this.

const ReadFromContract = useContractRead (
address....
onSuccess(data) {
   console.log('Contrad Data', data)
}
)

And then in your console look at the data it logged. You can then use that data as you wish.

**Reply by tmm** [reply]
Yup! Also, wagmi will infer types for you very soon (see https://github.com/wagmi-dev/wagmi/pull/941 powered by [ABIType](https://github.com/wagmi-dev/abitype)).
